test/profile_memory_usage.rb

```diff
@@ -0,0 +1,78 @@
+# frozen_string_literal: true
```

Comment:
This file should probably go into the `benchmark` directory since it is benchmarking the library?

Reply:
I put it here since it runs assertions on the `testcases` as well. But yes, you're right. The primary intention of the script is profiling after all.....